### PR TITLE
feat(medusa): handle product categories in import/export strategies

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- add ProductCategories to import and export strategies
- refactor ProductCategoriesService methods to use "retrieve_" pattern

---

RESOLVES CORE-1275

### CODE REVIEW

File: packages/medusa/src/services/product-category.ts [hunk 1]
```diff
@@ -110,25 +149,34 @@ class ProductCategoryService extends TransactionBaseService {
...
-    const query = buildQuery(selectors, config)
-    const productCategoryRepo = this.activeManager_.withRepository(
-      this.productCategoryRepo_
-    )
-
-    const productCategory = await productCategoryRepo.findOneWithDescendants(
-      query,
-      treeSelector
-    )
+    const selectors = Object.assign({}, { id: productCategoryId }, selector)
```
Comment (riqwan) [inline packages/medusa/src/services/product-category.ts]: added a redundant `{}` here

File: packages/medusa/src/strategies/batch-jobs/product/import.ts [hunk 2]
```diff
@@ -367,6 +380,33 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
...
+  private async processCategories(
+    data: { handle: string }[]
+  ): Promise<{ id: string }[]> {
+    const retIds: { id: string }[] = []
+    const transactionManager = this.transactionManager_ ?? this.manager_
+    const productCategoryService =
+      this.productCategoryService_.withTransaction(transactionManager)
+
+    for (const category of data) {
+      const categoryPartial = (await productCategoryService.retrieveByHandle(
```
Comment (riqwan) [inline packages/medusa/src/strategies/batch-jobs/product/import.ts:397]: q: should we try catch this bit?

**Review by riqwan — APPROVED** [review]
Clean! Lgtm. :)

File: packages/medusa/src/strategies/batch-jobs/product/import.ts (hunk 2 above)
Comment (fPolic) [reply packages/medusa/src/strategies/batch-jobs/product/import.ts:397]: we want this error to propagate to the caller function which then calls `ProductImportStrategy.throwDescriptiveError`

File: packages/medusa/src/services/product-category.ts [hunk 3]
```diff
@@ -91,10 +91,49 @@ class ProductCategoryService extends TransactionBaseService {
...
 
+  /**
+   * A generic retrieve for fining product categories by different attributes.
+   *
+   * @param config - the config of the product category to retrieve.
+   * @param selector
+   * @param treeSelector
+   * @return the product category.
+   */
+  protected async retrieve_(
```
Comment (olivermrbl) [inline packages/medusa/src/services/product-category.ts:102]: praise: Nice!

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**